npm/ng-packs/packages/permission-management/src/lib/components/permission-management.component.ts

```diff
@@ -153,15 +153,21 @@ export class PermissionManagementComponent
       this.permissions = this.permissions.map(per => {
         if (clickedPermission.name === per.name) {
           return { ...per, isGranted: !per.isGranted };
-        } else if (clickedPermission.name === per.parentName && clickedPermission.isGranted) {
-          return { ...per, isGranted: false };
+        } else if (clickedPermission.name === per.parentName) {
+          return { ...per, isGranted: !clickedPermission.isGranted };
         } else if (clickedPermission.parentName === per.name && !clickedPermission.isGranted) {
           return { ...per, isGranted: true };
         }
```

Comment:
I think we can remove this block

Reply:
The code: When a child is selected, parent is set as "selected"